javaparser-core/src/main/javacc/java.jj

```diff
@@ -711,36 +711,46 @@ StubUnit StubUnit():
     ModifierHolder modifier;
     TypeDeclaration tn = null;
     ModuleDeclaration module = null;
+    NodeList<AnnotationExpr> annotations = new NodeList<AnnotationExpr>();
+    AnnotationExpr annotation = null;
 }
 {
     ( LOOKAHEAD(2) ";" )*
-    [ LOOKAHEAD(PackageDeclaration()) pakage = PackageDeclaration() ]
+    /*
+     * Reading leading imports to support the old stub files that contain import declarations
+     * before the package declaration.
+     */
+    (in = ImportDeclaration() { imports = add(imports, in); })*
+
     /**
      * The stub file format permits having multiple classes and packages.
      * The packages are separated by a package statement: package my.package;.
      * Each package declaration may occur only once; in other words, all classes from a package must appear together.
      *
      * Going through a stub file and reading one compilation unit at a time
      * Number of compilation units in a stub file is 1+
-     */
-    (
-        /* Additional package declaration is not require, but still possible */
-        (pakage = PackageDeclaration())?
-
-        /*
-         * The only required import statements are the ones to import type annotations.
-         * Import statements for types are optional.
-         */
-        (in = ImportDeclaration() { imports = add(imports, in); })*
+     * Each CompilationUnit declaration starts with a package declaration.
+    */
+    (pakage = PackageDeclaration())?
+    (in = ImportDeclaration() { imports = add(imports, in); })*
 
-        {
-            types = emptyList();
-            modifier = null;
-            tn = null;
-            module = null;
-        }
-
-            (
+    (LOOKAHEAD(3, PackageDeclaration())
+        (
+             {
+                cu = new CompilationUnit(range(token_source.getHomeToken(), token()), pakage, imports, types, module);
+                compilationUnits.add(cu);
+                types = emptyList();
+                tn = null;
+                module = null;
+             }
+             pakage = PackageDeclaration()
+             (in = ImportDeclaration() { imports = add(imports, in); })*
```

Comment:
According to the Checker Framework manual, all imports declaration should be at the beginning of a stub file. 
https://checkerframework.org/manual/#stub
Other imports readings were added because some of the legacy stub files have imports declarations between the compilation units.
I am not sure how does it affect the conflict.
However, in the old stubparser, it also collects all the imports into a single list.
It could be left like that or I can remove redundant imports readings and change all the legacy stub files to declare imports only at the beginning of files.